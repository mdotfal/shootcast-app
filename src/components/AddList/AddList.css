.addlist-form {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
};

input {
  height: 14px;;
}
