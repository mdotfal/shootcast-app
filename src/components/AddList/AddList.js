import React, { Component } from 'react';
import AppContext from '../../AppContext';
import config from '../../config';
import './AddList.css';

class AddList extends Component {

  static contextType = AppContext;

  addList = event => {
  
    const name = event.target.inputList.value;

    fetch( `${ config.API_ENDPOINT }/api/lists`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify({ name })
    })
    .then( res => res.json())
    .then( data => {
      this.context.onAddList( data );
      event.target.inputList.value = '';
    })
    .catch( error => { 
      console.log( error );
    });
  }

  onSubmit = e => {
    e.preventDefault();
    this.addList( e );
  }

  render() {
    return (
      <div className="addlist">
        <form className="addlist-form" onSubmit={ this.onSubmit }>
          <label htmlFor="inputList">
            *
          </label>
          <input type="text" placeholder="Southern California" name="inputList" id="inputList" required />
          <button
            type="submit" 
          >Add List</button>
        </form>
      </div>
    );
  };
};

export default AddList;