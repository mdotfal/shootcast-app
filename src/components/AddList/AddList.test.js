import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddList from './AddList';

describe( `AddList component`, () => {
  it(Â `renders without errors`, () => {
    const wrapper = shallow( <AddList /> )
    expect( toJson( wrapper )).toMatchSnapshot()
  })
})