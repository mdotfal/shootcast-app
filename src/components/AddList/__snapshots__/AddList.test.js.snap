// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddList component renders without errors 1`] = `
<div
  className="addlist"
>
  <form
    className="addlist-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="inputList"
    >
      *
    </label>
    <input
      id="inputList"
      name="inputList"
      placeholder="Southern California"
      required={true}
      type="text"
    />
    <button
      type="submit"
    >
      Add List
    </button>
  </form>
</div>
`;
